src/app/clusters/resource-monitoring-server/resource-monitoring-server.h

```diff
@@ -0,0 +1,153 @@
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandlerInterface.h>
+#include <app/clusters/resource-monitoring-server/resource-monitoring-cluster-objects.h>
+#include <app/util/af.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ResourceMonitoring {
+
+class Instance : public CommandHandlerInterface, public AttributeAccessInterface
+{
+
+public:
+    void LoadPersistentAttributes();
+
+    /**
+     * Initialise the Resource Monitoring cluster.
+     *
+     * @return CHIP_ERROR       If the cluster ID given ist not a valid Resource Monitoring cluster ID.
+     * @return CHIP_ERROR       If the endpoint and cluster ID have not been enabled in zap.
+     * @return CHIP_ERROR       If the CommandHandler or Attribute Handler could not be registered.
+     * @return CHIP_ERROR       If the AppInit() function returned an error.
+     *
+     * @return CHIP_NO_ERROR    If the cluster was initialised successfully.
+     */
+    CHIP_ERROR Init();
+
+    // CommandHandlerInterface
+    void InvokeCommand(HandlerContext & ctx) override;
+    CHIP_ERROR EnumerateAcceptedCommands(const ConcreteClusterPath & cluster, CommandIdCallback callback, void * context) override;
+
+    /**
+     * returns true if the given feature is supported by the cluster.
+     * @param feature   The aFeature to check.
+     */
+    bool HasFeature(ResourceMonitoring::Feature aFeature) const;
+
+    // AttributeAccessInterface
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+    CHIP_ERROR Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder) override;
+
+    // Attribute setters
+    chip::Protocols::InteractionModel::Status UpdateCondition(uint8_t aNewCondition);
+    chip::Protocols::InteractionModel::Status UpdateChangeIndication(ChangeIndicationEnum aNewChangeIndication);
+    chip::Protocols::InteractionModel::Status UpdateInPlaceIndicator(bool aNewInPlaceIndicator);
+    chip::Protocols::InteractionModel::Status UpdateLastChangedTime(DataModel::Nullable<uint32_t> aNewLastChangedTime);
+
+    // Attribute getters
+    uint8_t GetCondition() const;
+    ChangeIndicationEnum GetChangeIndication() const;
+    DegradationDirectionEnum GetDegradationDirection() const;
+    bool GetInPlaceIndicator() const;
+    DataModel::Nullable<uint32_t> GetLastChangedTime() const;
+
+private:
+    EndpointId mEndpointId{};
+    ClusterId mClusterId{};
+
+    // attribute Data Store
+    chip::Percent mCondition                             = 100;
+    const DegradationDirectionEnum mDegradationDirection = DegradationDirectionEnum::kDown;
+    ChangeIndicationEnum mChangeIndication               = ChangeIndicationEnum::kOk;
+    bool mInPlaceIndicator                               = true;
+    DataModel::Nullable<uint32_t> mLastChangedTime;
+
+    uint32_t mFeature;
+
+    const bool mResetCondtitionCommandSupported = false;
```

Comment:
nit: typo `mResetConditionCommandSupported`

Also, do NOT have const members for something that may improve in the future.

Reply:
fixed